Client/FeatureFlags/NimbusFlaggableFeature.swift

```diff
@@ -0,0 +1,163 @@
+        self.featureID = featureID
+        self.profile = profile
+    }
+
+    // MARK: - Public methods
+    public func isNimbusEnabled(using nimbusLayer: NimbusFeatureFlagLayer) -> Bool {
+        let nimbusValue = nimbusLayer.checkNimbusConfigFor(featureID)
+
+        switch featureID {
+        case .pocket:
+            return nimbusValue && Pocket.IslocaleSupported(Locale.current.identifier)
+        default:
+            return nimbusValue
+        }
+    }
+
+    /// Returns whether or not the feature's state was changed by the user. If no
+    /// preference exists, then the underlying Nimbus default is used. If a specific
+    /// setting is required (ie. startAtHome, which has multiple types of setting),
+    /// then we should be using `getPreferenceFor`
+    public func isUserEnabled(using nimbusLayer: NimbusFeatureFlagLayer) -> Bool {
+        guard let optionsKey = featureKey,
+              let option = profile.prefs.boolForKey(optionsKey)
+        else { return isNimbusEnabled(using: nimbusLayer) }
+
+        return option
+    }
+
+    /// Returns the feature option represented as a String. The `FeatureFlagManager` will
+    /// convert it to the appropriate type.
+    public func getUserPreference(using nimbusLayer: NimbusFeatureFlagLayer) -> String? {
+        if let optionsKey = featureKey,
+           let existingOption = profile.prefs.stringForKey(optionsKey) {
+            return existingOption
+        }
+
+        switch featureID {
+        case .startAtHome:
+            return nimbusLayer.checkNimbusConfigForStartAtHome().rawValue
+
+        default:
+            return nil
+        }
+    }
+
+    /// Set a user preference that is of type on/off, to that respective state.
+    ///
+    /// Not all features are user togglable. If there exists no feature key - as defined
+    /// in the `featureKey()` function - with which to write to UserDefaults, then the
+    /// feature cannot be turned on/off.
+    public func setUserPreference(to state: Bool) {
+        guard let key = featureKey else { return }
+
+        profile.prefs.setBool(state, forKey: key)
+    }
+
+    /// Allows to directly set the state of a feature using a string to allow for
+    /// states beyond on and off.
+    ///
+    /// Not all features are user togglable. If there exists no feature key - as defined
+    /// in the `featureKey()` function - with which to write to UserDefaults, then the
+    /// feature cannot be turned on/off.
+    public func setUserPreference(to option: String) {
+        guard !option.isEmpty,
+              let optionsKey = featureKey
+        else { return }
+
+        switch featureID {
+        case .startAtHome:
+            if option == StartAtHomeSetting.afterFourHours.rawValue
+                || option == StartAtHomeSetting.always.rawValue {
+                setUserPreference(to: true)
+            } else {
+                setUserPreference(to: false)
+            }
```

Comment:
nit, just a question of preference
```suggestion
            let userPreference = option == StartAtHomeSetting.afterFourHours.rawValue || option == StartAtHomeSetting.always.rawValue
            setUserPreference(to: userPreference)
```